### PR TITLE
HRCPP-387 Extend test suite for DIGEST-MD5 and PLAIN authentication

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.jboss.org/browse/HRCPP-387

### CODE REVIEW

File: src/test/cs/Infinispan/HotRod/BaseAuthorizationTest.cs [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+            AssertError(writerCache, cache => TestContainsKey(cache));
+            AssertError(writerCache, cache => TestGetNonExistent(cache));
+            AssertError(writerCache, cache => TestGetVersioned(cache));
+            AssertError(writerCache, cache => TestGetWithMetadata(cache));
+        }
+
+        [Test]
+        public void WriterPerformsSupervisorOpsTest()
+        {
+            //test only minimum of operations because of JIRA: TBD
```
Comment (mgencur) [inline src/test/cs/Infinispan/HotRod/BaseAuthorizationTest.cs]: HRCPP-388

File: src/test/resources/clustered-sasl-cs.xml [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                        <authorization enabled="true" roles="admin reader writer supervisor"/>
+                    </security>
+                </distributed-cache>
+            </cache-container>
+        </subsystem>
+        <subsystem xmlns="urn:infinispan:server:endpoint:9.0">
+            <hotrod-connector socket-binding="hotrod" cache-container="clustered">
+                <topology-state-transfer lazy-retrieval="false" lock-timeout="1000" replication-timeout="5000"/>
+                <authentication security-realm="ApplicationRealm">
+                    <sasl mechanisms="DIGEST-MD5 PLAIN" qop="auth" server-name="${server.name:node0}" strength="high medium low">
```
Comment (rigazilla) [inline src/test/resources/clustered-sasl-cs.xml]: sasl server-name attribute must be the same for all the cluster nodes.
server-name="node0" should be fine

Comment (mgencur) [inline src/test/resources/clustered-sasl-cs.xml]: OK. That explains a lot:)

**Comment by mgencur** [comment]
Fixed. Now both servers have the name equal to "node0". Also, closing https://issues.jboss.org/browse/HRCPP-386

**Comment by rigazilla** [comment]
merged thanks!

**Pull request merged**